src/main/java/core/stack/MateStack.java

```diff
@@ -18,10 +20,16 @@ public void push(T value) {
     }
 
     public T peek() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
         return stack[size - 1];
     }
 
     public T pop() {
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
         T value = stack[size - 1];
         stack[size-- - 1] = null;
```

Comment:
Strange decided, i hope mentor will like it :)

Reply:
I know, but I like it)